pkg/liqo-controller-manager/storageprovisioner/remoteprovisioner.go

```diff
@@ -13,3 +13,104 @@
 // limitations under the License.
 
 package storageprovisioner
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1apply "k8s.io/client-go/applyconfigurations/core/v1"
+	corev1clients "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/sig-storage-lib-external-provisioner/v7/controller"
+
+	"github.com/liqotech/liqo/pkg/virtualKubelet/forge"
+)
+
+// ProvisionRemotePVC ensures the existence of a remote PVC and returns a virtual PV for that remote storage device.
+func ProvisionRemotePVC(ctx context.Context,
+	options controller.ProvisionOptions,
+	remoteNamespace, remoteRealStorageClass string,
+	remotePvcLister corev1listers.PersistentVolumeClaimNamespaceLister,
+	remotePvcClient corev1clients.PersistentVolumeClaimInterface) (*v1.PersistentVolume, controller.ProvisioningState, error) {
+	virtualPvc := options.PVC
+
+	mutation := remotePersistentVolumeClaim(virtualPvc, remoteRealStorageClass, remoteNamespace)
+	_, err := remotePvcClient.Apply(ctx, mutation, forge.ApplyOptions())
+	if err != nil {
+		return nil, controller.ProvisioningInBackground, err
+	}
+
+	pv := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: options.PVName,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			StorageClassName: options.StorageClass.Name,
+			AccessModes:      options.PVC.Spec.AccessModes,
+			Capacity: v1.ResourceList{
+				v1.ResourceStorage: options.PVC.Spec.Resources.Requests[v1.ResourceStorage],
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/tmp/liqo-placeholder",
+				},
+			},
+			NodeAffinity: &v1.VolumeNodeAffinity{
+				Required: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      "kubernetes.io/hostname",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{options.SelectedNode.Name},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if options.StorageClass.ReclaimPolicy != nil {
+		pv.Spec.PersistentVolumeReclaimPolicy = *options.StorageClass.ReclaimPolicy
+	}
+
+	return pv, controller.ProvisioningFinished, nil
+}
+
+// remotePersistentVolumeClaim forges the apply patch for the reflected PersistentVolumeClaim, given the local one.
+func remotePersistentVolumeClaim(virtualPvc *v1.PersistentVolumeClaim,
+	storageClass, namespace string) *v1apply.PersistentVolumeClaimApplyConfiguration {
+	return v1apply.PersistentVolumeClaim(virtualPvc.Name, namespace).
+		WithLabels(virtualPvc.GetLabels()).
+		WithLabels(forge.ReflectionLabels()).
```

Comment:
Is it meaningful to propagate also annotations?

Reply:
No, annotations are used to mark the state of the volume (bound or not, on which node, ...) that have no meaning in the remote cluster